src/supervisor/watchers/handlers/index.ts

```diff
@@ -319,7 +133,88 @@ export async function setupInformer(
           FALSY_WORKLOAD_NAME_MARKER;
         logger.warn(
           { ...logContext, error, name },
-          'could not execute the informer handler for a workload',
+          'could not execute the namespaced informer handler for a workload',
+        );
+      }
+    });
+  }
+
+  await informer.start();
+}
+
+export async function setupClusterInformer(
+  workloadKind: WorkloadKind,
+): Promise<void> {
+  const logContext: Record<string, unknown> = { workloadKind };
+  const isSupported = await isSupportedClusterWorkload(workloadKind);
+  if (!isSupported) {
+    logger.debug(
+      logContext,
+      'The Kubernetes cluster does not support this workload',
+    );
+    return;
+  }
+
+  const workloadMetadata = workloadWatchMetadata[workloadKind];
+  const endpoint = workloadMetadata.clusterEndpoint;
+
+  const listMethod = workloadMetadata.clusterListFactory();
+  const loggedListMethod = async () => {
+    try {
+      return await kubernetesApiWrappers.retryKubernetesApiRequest(() =>
+        listMethod(),
+      );
+    } catch (err) {
+      logger.error(
+        { ...logContext, err },
+        'error while listing workloads in cluster',
+      );
+      throw err;
+    }
+  };
+
+  const informer = makeInformer<KubernetesObject>(
+    kubeConfig,
+    endpoint,
+    loggedListMethod,
+  );
+
+  informer.on(ERROR, (err) => {
+    // Types from client library insists that callback is of type KubernetesObject
+    const code = (err as any).code || '';
+    if (RETRYABLE_NETWORK_ERRORS.includes(code)) {
+      logger.debug(
+        logContext,
+        `informer ${code} occurred, restarting informer`,
```

Comment:
should we be using string interpolation in error messages?

Reply:
good point, we shouldn't!